icu4c/source/test/cintltst/capitst.c

```diff
@@ -2565,4 +2566,18 @@ static void TestStrcollNull(void) {
     ucol_close(coll);
 }
 
+static void TestBogusLocaleID(void) {
+    UErrorCode err = U_ZERO_ERROR;
+    UCollator* c1 = ucol_open("en-US-u-kn-true", &err);
+    UCollator* c2 = ucol_open("en_US-u-kn-true", &err);
+    
+    if (assertSuccess("Failed to create collators", &err)) {
+        assertTrue("Comparison using \"normal\" collator failed", !ucol_greater(c1, u"2", -1, u"10", -1));
```

Comment:
Double negative... Weird that we don't have a ucol_less() function, but I would assertFalse(..., ucol_greater(...))

Reply:
There doesn't seem to be an assertFalse() function in cintltst.c.

Reply:
optional(!): You could add assertFalse() :-)